tests/msim/test_msim_ACTIONX.cpp

```diff
@@ -0,0 +1,154 @@
+
+#include <boost/test/unit_test.hpp>
+
+#include <ert/util/test_work_area.h>
+
+#include <opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/SummaryState.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Action/ActionAST.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Action/ActionContext.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Action/Actions.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Action/ActionX.hpp>
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/Parser/ParseContext.hpp>
+#include <opm/parser/eclipse/Parser/ErrorGuard.hpp>
+
+#include <opm/output/eclipse/EclipseIO.hpp>
+
+#include <opm/msim/msim.hpp>
+
+using namespace Opm;
+
+
+
+struct test_data {
+    Deck deck;
+    EclipseState state;
+    Schedule schedule;
+    SummaryConfig summary_config;
+
+    test_data(const std::string& deck_string) :
+        deck( Parser().parseString(deck_string)),
+        state( this->deck ),
+        schedule( this->deck, this->state.getInputGrid(), this->state.get3DProperties(), this->state.runspec()),
+        summary_config( this->deck, this->schedule, this->state.getTableManager())
+    {
+    }
+};
+
+double prod_opr(const EclipseState&  es, const Schedule& sched, const data::Solution& sol, size_t report_step, double seconds_elapsed) {
+    const auto& units = es.getUnits();
+    double oil_rate = 1.0;
+    return -units.to_si(UnitSystem::measure::rate, oil_rate);
+}
+
+double prod_wpr_P1(const EclipseState&  es, const Schedule& sched, const data::Solution& sol, size_t report_step, double seconds_elapsed) {
+    const auto& units = es.getUnits();
+    double water_rate = 0.0;
+    return -units.to_si(UnitSystem::measure::rate, water_rate);
+}
+
+double prod_wpr_P2(const EclipseState&  es, const Schedule& sched, const data::Solution& sol, size_t report_step, double seconds_elapsed) {
+    const auto& units = es.getUnits();
+    double water_rate = 0.0;
+    if (report_step > 5)
+        water_rate = 2.0;  // => WWCT = WWPR / (WOPR + WWPR) = 2/3
+
+    return -units.to_si(UnitSystem::measure::rate, water_rate);
+}
+
+double prod_wpr_P3(const EclipseState&  es, const Schedule& sched, const data::Solution& sol, size_t report_step, double seconds_elapsed) {
+    const auto& units = es.getUnits();
+    double water_rate = 0.0;
+    return -units.to_si(UnitSystem::measure::rate, water_rate);
+}
+
+double prod_wpr_P4(const EclipseState&  es, const Schedule& sched, const data::Solution& sol, size_t report_step, double seconds_elapsed) {
+    const auto& units = es.getUnits();
+    double water_rate = 0.0;
+    if (report_step > 10)
+        water_rate = 2.0;
+
+    return -units.to_si(UnitSystem::measure::rate, water_rate);
+}
+
+
+BOOST_AUTO_TEST_CASE(WELL_CLOSE_EXAMPLE) {
+#include "actionx1.include"
```

Comment:
Is this file supposed to be automatically generated?  If not, it is missing in the fileset.

Reply:
Added now - thanks